docker/api/service.py

```diff
@@ -0,0 +1,62 @@
+from ..utils import check_resource
+
+
+class ServiceApiMixin(object):
+    def services(self, name=None, partial_id=None):
+        url = self._url("/services")
+        params = {
+            'name': name,
+            'partial-id': partial_id,
+        }
+        res = self._get(url, params=params)
+        return self._result(res, json=True) or []
+
+    def publish_service(self, name, network_name=None):
+        data = {
+            'name': name,
+            'network_name': network_name,
+        }
+        url = self._url("/services")
+        res = self._post_json(url, data=data)
+        return self._result(res, json=True)
+
+    def unpublish_service(self, service_id):
+        url = self._url("/services/{0}".format(service_id))
+        res = self._delete(url)
+        self._raise_for_status(res)
+
+    def inspect_service(self, service_id):
+        url = self._url("/services/{0}".format(service_id))
+        res = self._get(url)
+        return self._result(res, json=True)
+
+    def service_backends(self, service_id):
+        url = self._url("/services/{0}/backend".format(service_id))
+        res = self._get(url)
+        return self._result(res, json=True)
+
+    def sandboxes(self, partial_container_id=None):
+        url = self._url("/sandboxes")
+        params = {}
+        if partial_container_id:
+            params['partial-container-id'] = partial_container_id
+        res = self._get(url, params=params)
+        return self._result(res, json=True) or []
+
+    @check_resource
+    def attach_container_to_service(self, container, service_id):
+        sandbox = self.sandboxes(partial_container_id=container)[0]
```

Comment:
the apis are being simplified and as @shin- indicated, there will be a /connect api with container as the resource of interest. sandbox is a backend detail and will not be exposed in the API.

This API that @aanand is using is based on experimental version and will be changed soon.
I will add an update here once the docker/docker API/UX change PR is pushed.